### PR TITLE
Fix underpriced transaction

### PR DESCRIPTION
**Pull request by hackaugusto**
Using the txpool api to get the used nonces because getTransactionCount does not include the transactions from the pool.

### CODE REVIEW

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -32,6 +32,28 @@
...
+    queued = pool.get('queued', {}).get(address)
+    if queued:
+        return max(queued.keys()) + 1
+
+    pending = pool.get('pending', {}).get(address)
+    if pending:
+        return max(pending.keys()) + 1
+
+    # The first valid nonce is 0, therefore the count is already the next
+    # avaiable nonce
```
Comment (palango) [inline raiden/network/rpc/client.py]: s/avaiable/available

File: raiden/network/rpc/client.py [hunk 2]
```diff
@@ -32,6 +32,28 @@
 log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
 
 
+def discover_next_available_nonce(web3, address):
+    """Returns the next available nonce for `address`."""
+
+    # The nonces of the mempool transactions are considered used, and it's
+    # assumed these transactions are different from the ones currenlty pending
```
Comment (palango) [inline raiden/network/rpc/client.py]: s/currenlty/currently

**Comment by hackaugusto** [comment]
This has to be tested with parity

File: raiden/network/rpc/client.py [hunk 3]
```diff
@@ -32,6 +32,36 @@
...
+    if queued:
+        return max(queued.keys()) + 1
+
+    pending = pool.get('pending', {}).get(address)
+    if pending:
+        return max(pending.keys()) + 1
+
+    # The first valid nonce is 0, therefore the count is already the next
+    # available nonce
+    return web3.eth.getTransactionCount(to_checksum_address(address), 'latest')
```
Comment (rakanalh) [inline raiden/network/rpc/client.py]: `address` is already being checksummed when calling this function so calling it here on the address again is redundant.

**Comment by LefterisJP** [comment]
@hackaugusto I checked and it seems the `txpool` is a geth only construct. But parity has `parity_nextNonce`. Unfortunately it's not offered by web3.py but we can simply make a json rpc call to parity. I added a commit that does that. Needs to actually be tested with parity before merging.

**Comment by hackaugusto** [comment]
> While a RecoverableError might not be a direct issue as it is expected, would you think this might have any unexpected side effects?

Well, it's possible, but the alternative would to match the pending transaction to the transaction in the pool, and that is way more work (I'm not saying we should not do it, but this fix allows fast restarts now)

**Comment by palango** [comment]
This broke infura: See https://github.com/raiden-network/raiden/issues/2951 and messages on gitter.

**Pull request merged**